File: Interviews/Striver-SDE-Sheet/Day-17-Binary-Tree/Inorder-Traversal/94.cpp
void traverse(TreeNode* root, vector<int> &ans) {
    if (!root) return;
    traverse(root->left, ans);
    ans.push_back(root->val);
    traverse(root->right, ans);
}

vector<int> inorderTraversal(TreeNode* root) {
    vector<int> ans;
    traverse(root, ans);
    return ans;
}

File: Interviews/Striver-SDE-Sheet/Day-17-Binary-Tree/Inorder-Traversal/94.py
class Solution:
    def traverse(self, root, ans):
        if not root: return

        self.traverse(root.left, ans)
        ans.append(root.val)
        self.traverse(root.right, ans)

    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        self.traverse(root, ans)
        return ans
